include/deal.II/grid/manifold.h

```diff
@@ -32,13 +33,75 @@ DEAL_II_NAMESPACE_OPEN
 // forward declaration
 template <int, typename> class Table;
 
+namespace internal
+{
+  /**
+   * The function Manifolds::get_default_points_and_weights returns an array
+   * of points whose length depends on the dimension of the current
+   * geometrical structure. We obtain this value at compile time with a
+   * sequence of specialized template <code>constexpr</code> functions.
+   */
+  template<int dim>
+  inline
+  constexpr std::size_t n_default_points_per_cell()
+  {
+    return std::numeric_limits<std::size_t>::max();
```

Comment:
There's a cool construct you can use for this general template (which you really don't want to make available to begin with):
```
template<int dim>
constexpr std::size_t n_default_points_per_cell() = delete;
```
This declares the general template (and consequently allows for specializations) but immediately deletes it again.

It's about as useful this way as if you had put a `static_assert(false);` into the body of the function :-)

Reply:
This is a neat trick. I remember reading about `=delete`ing general functions and this is exactly the place to do it :)